prime-dotnet-webapi/Startup.cs

```diff
@@ -251,6 +251,8 @@ public void Configure(IApplicationBuilder app, IWebHostEnvironment env, IHostApp
             // Matches request to an endpoint
             app.UseRouting();
 
+            app.UseMiddleware<RequestLoggingMiddleware>();
+

```

Comment:
Are you certain this is the correct location to register the middleware? Middleware are invoked in a pipeline in the order they are added to the app, so where you put `app.UseMiddleware` matters. If you look at the line below this, it mentions "Make sure to put this middleware right after `UseRouting()`." so I am unsure that this is the correct place for it.